pkg/controller/networkpolicy/reconciler.go

```diff
@@ -282,26 +323,7 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToPublicNetworks(ctx context.Con
 		policy.Spec = networkingv1.NetworkPolicySpec{
 			PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.LabelNetworkPolicyToPublicNetworks: v1beta1constants.LabelNetworkPolicyAllowed}},
 			Egress: []networkingv1.NetworkPolicyEgressRule{{
-				To: []networkingv1.NetworkPolicyPeer{{
-					IPBlock: &networkingv1.IPBlock{
-						CIDR: "0.0.0.0/0",
-						Except: append(
-							toCIDRStrings(allPrivateNetworkBlocksV4()...),
-							r.RuntimeNetworks.BlockCIDRs...,
-						),
-					},
-				}, {
-					IPBlock: &networkingv1.IPBlock{
-						CIDR: "::/0",
-						// TODO: exclude private blocks so that this actually allows only public networks
-						// Except: append([]string{
-						// 	private8BitBlock().String(),
-						// 	private12BitBlock().String(),
-						// 	private16BitBlock().String(),
-						// 	carrierGradeNATBlock().String(),
-						// }, r.SeedNetworks.BlockCIDRs...),
-					},
-				}},
+				To: append(peersV4, peersV6...),
```

Comment:
@ScheererJ @axel7born @DockToFuture
We tried our best to implement the `allow-to-public-networks` policy for IPv6 but we are not sure about the details.
Can you take a look at this commit to see if our implementation makes sense?